### PR TITLE
Initial implementation and tests setup

### PR DESCRIPTION
**Pull request by momegas**
The approach in the setup is to mock requests and test the SDK this way.
The reason is that `TestClient` does some magic and does not run the API exposing the host and port. I found it quite complex to run the API like this so I went with the simple way of mocking the back end.

### CODE REVIEW

File: src/sdk/whitebox.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            json=new_model.dict(),
+            headers={"api-key": self.api_key},
+        )
+
+        logger.info(result.json())
+        return result.json()
+
+    def get_model(self, model_id: str):
+        result = requests.get(
+            url=f"{self.host}/v1/models/{model_id}", headers={"api-key": self.api_key}
```
[the remark this answers is not in the document]
Comment (momegas) [reply src/sdk/whitebox.py:48]: You mean the API version?

[the remark this answers is not in the document]
Comment (gcharis) [reply src/sdk/whitebox.py:49]: yep

[the remark this answers is not in the document]
Comment (momegas) [reply src/sdk/whitebox.py:55]: I think the right way to do this is through `experimental` functions or something like this.

[the remark this answers is not in the document]
Comment (gcharis) [reply src/sdk/whitebox.py:55]: what do you mean?

File: src/sdk/whitebox.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+    def delete_model(self, model_id: str):
+        """
+        Deletes a model by its id. If any error occurs, returns False.
+        """
+        result = requests.delete(
+            url=f"{self.host}/v1/models/{model_id}", headers={"api-key": self.api_key}
+        )
+
+        if result.status_code == status.HTTP_200_OK:
```
Comment (gcharis) [inline src/sdk/whitebox.py:70]: I would go for 204, wydt?

File: src/sdk/whitebox.py [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+    ):
+        """
+        Logs a training dataset for a model.
+
+        Non processed is a dataframe with the raw data.
+        Processed is a dataframe with the data after it has been processed and before it has entered the model.
+        """
+        non_processed_json = non_processed.to_dict(orient="records")
+        processed_json = processed.to_dict(orient="records")
+
```
Comment (gcharis) [inline src/sdk/whitebox.py:86]: Maybe a condition to check the lengths are the same?

File: src/sdk/whitebox.py [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+    ):
+        """
+        Logs a inferences of a model.
+
+        Non processed is a dataframe with the raw data.
+        Processed is a dataframe with the data after it has been processed and before it has entered the model.
+        """
+        non_processed_json = non_processed.to_dict(orient="records")
+        processed_json = processed.to_dict(orient="records")
+
```
Comment (gcharis) [inline src/sdk/whitebox.py:122]: Same? Check the lengths?

File: src/sdk/whitebox.py (hunk 3 above)
Comment (momegas) [reply src/sdk/whitebox.py:86]: Good one ;)

File: src/sdk/whitebox.py (hunk 2 above)
Comment (momegas) [reply src/sdk/whitebox.py:70]: Its already 200 in the API and didnt want to touch other things for now.
I want to see this in another iteration. 
There are a lot of things that need changing, im afraid.

**Review by sinnec — APPROVED** [review]
Guess it's OK. I really need to get in depth with that SDK thing.
P.S.: Nice changing "many" to "batch" in inferences API. Was on my mind for quite some time.

File: src/sdk/whitebox.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (momegas) [reply src/sdk/whitebox.py]: Done

**Review by gcharis — APPROVED** [review]
Looks good 👍

**Pull request merged**